src/lib/spinel/radio_spinel.cpp

```diff
@@ -997,10 +997,15 @@ otError RadioSpinel::AddSrcMatchShortEntry(uint16_t aShortAddress)
 {
     otError error;
 
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
+    EXPECT_NO_ERROR(error = (mSrcMatchShortEntryCount < OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES)
+                                ? OT_ERROR_NONE
+                                : OT_ERROR_NO_BUFS);
+#endif
```

Comment:
I don't think this error is processed on the caller side. If this happens (i.e., we cannot save the request in the source match table list for RCP restoration), RCP restoration behavior may malfunction and break.

Ideally, `OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES` should be large enough to accommodate the maximum number of children supported by the device so we should not get to this state. If we do, asserting may be better than returning an error and proceeding, which could create unintended behavior.

Thoughts?